### PR TITLE
Improve README syntax highlight example

### PR DESCRIPTION
**Pull request by dlqqq**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

Improves README syntax highlighting example. The current example produces a TypeScript error by default. See here: https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/479#issuecomment-1267772279

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -354,14 +354,14 @@ ReactDom.render(
         const match = /language-(\w+)/.exec(className || '')
         return !inline && match ? (
           <SyntaxHighlighter
+            {...props}
```
Comment (remcohaszing) [inline readme.md:357]: It might be better to not even pass spread props at all.

Anyway, I'm not necessarily against it.

File: readme.md [hunk 2]
```diff
@@ -354,14 +354,14 @@ ReactDom.render(
...
             children={String(children).replace(/\n$/, '')}
             style={dark}
             language={match[1]}
             PreTag="div"
-            {...props}
           />
         ) : (
-          <code className={className} {...props}>
+          <code {...props} className={className}>
```
Comment (remcohaszing) [inline readme.md:364]: The order doesn't matter here.

**Review by remcohaszing — APPROVED** [review]

**Review by ChristianMurphy — APPROVED** [review]
Thanks @dlqqq!
I'd second @remcohaszing's suggestions.

File: readme.md (hunk 2 above)
Comment (wooorm) [reply readme.md:364]: `true`, but its not bad to be consistent with the other example!

File: readme.md (hunk 1 above)
Comment (wooorm) [reply readme.md:357]: I think there was something about `SyntaxHighlighter` passing all other props through on the `div` or so:

> spread props pass arbitrary props to pre tag wrapping code.
> — https://github.com/react-syntax-highlighter/react-syntax-highlighter/tree/efc3f7b7537d1729193b7a472067bcbe6cbecaf1#props

Could be removed too, okay with me. But that can be a different PR!

**Review by wooorm — APPROVED** [review]

**Pull request merged**